noir-projects/aztec-nr/aztec/src/note/note_metadata.nr

```diff
@@ -40,7 +40,7 @@ global NoteStage: NoteStageEnum =
 /// This represents a note in any of the three valid stages (pending same phase, pending previous phase, or settled). In
 /// order to access the underlying fields callers must first find the appropriate stage (e.g. via `is_settled()`) and
 /// then convert this into the appropriate type (e.g. via `to_settled()`).
-#[derive(Eq, Serialize, Packable)]
+#[derive(Deserialize, Eq, Serialize, Packable)]
```

Comment:
We now return `RetrievedNote` from a contract function which forced me to do this change.